micro-rdk/src/common/conn/viam.rs

```diff
@@ -86,15 +89,28 @@ impl From<&proto::app::v1::CloudConfig> for RobotCloudConfig {
     }
 }
 
+#[cfg(not(feature = "ota"))]
 pub trait ViamServerStorage:
     RobotConfigurationStorage + WifiCredentialStorage + Clone + 'static
 {
 }
+#[cfg(not(feature = "ota"))]
 impl<T> ViamServerStorage for T where
     T: RobotConfigurationStorage + WifiCredentialStorage + Clone + 'static
 {
 }
 
+#[cfg(feature = "ota")]
+pub trait ViamServerStorage:
+    RobotConfigurationStorage + WifiCredentialStorage + OtaMetadataStorage + Clone + 'static
+{
+}
+#[cfg(feature = "ota")]
+impl<T> ViamServerStorage for T where
+    T: RobotConfigurationStorage + WifiCredentialStorage + OtaMetadataStorage + Clone + 'static
+{
+}
+
```

Comment:
not a fan of how this looks, open to suggestions.

Reply:
curious, what bothers you about it?

Reply:
just having 4 things that look very similar and all the feature-gating. nothing functional